lib/logstash/outputs/sns.rb

```diff
@@ -70,54 +63,45 @@ def register
   def receive(event)
     return unless output?(event)
 
-    arn     = Array(event["sns"]).first || @arn
-
-    raise "An SNS ARN required." unless arn
-
-    message = Array(event["sns_message"]).first
-    subject = Array(event["sns_subject"]).first || event["host"]
-
-    # Ensure message doesn't exceed the maximum size.
-    if message
-      # TODO: Utilize `byteslice` in JRuby 1.7: http://jira.codehaus.org/browse/JRUBY-5547
-      message = message.slice(0, MAX_MESSAGE_SIZE_IN_BYTES)
+    if (sns_msg = Array(event["sns_message"]).first)
+      send_sns_message(event_arn(event), event_subject(event), sns_msg)
```

Comment:
@ph I'm kind of confused by this myself. I didn't realize that any field in logstash could be an array. Is this code correct in this? Should it behave this way at all?